### PR TITLE
setup: fix package versioning

### PR DESCRIPTION
**Pull request by pazembrz**
Allow autosemver correctly manage versions of packages

### CODE REVIEW

**Review by MJedr — APPROVED** [review]

**Review by benjamin-bergia — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Cache Python
+        uses: actions/cache@master
+        id: cache
+        with:
+          path: ${{ env.pythonLocation }}
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply .github/workflows/build-and-release.yml:32]: This is set by actions/setup-python@v2 step already.

File: .github/workflows/build-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ github.sha }}
+
+      - name: Install python dependencies
+        run: |
+          ./install.sh
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply .github/workflows/build-and-release.yml:37]: :+1:

File: .github/workflows/build-and-release.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+      - name: Show python dependencies
+        run: |
+          pip freeze
+
+      - name: Run tests
+        run: |
+          ./run-tests.sh
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply .github/workflows/build-and-release.yml:45]: :+1:

File: .github/workflows/build-and-release.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+      - name: Build package
+        run: |
+          python setup.py sdist bdist_wheel
+
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@v1.3.1
+        with:
+          user: __token__
+          password: ${{ secrets.pypi_password }}
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply .github/workflows/build-and-release.yml:99]: Haris did this already from what I know

[the remark this answers is not in the document]
Comment (benjamin-bergia) [reply .github/workflows/build-and-release.yml:99]:
```suggestion
          password: ${{ secrets.PYPI_PASSWORD }}
```
I checked and the secret is already there

File: .github/workflows/build-and-release.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+  Release:
+    if: ${{ github.event_name == 'push' }}
+    needs: Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          python-version: 2.7
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply .github/workflows/build-and-release.yml:72]: :+1:

File: .github/workflows/build-and-release.yml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [2.7, 3.6]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          python-version: ${{ matrix.python-version }}
```
[the remark this answers is not in the document]
Comment (pazembrz) [reply .github/workflows/build-and-release.yml:20]: :+1:

File: .github/workflows/build-and-release.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (pazembrz) [reply .github/workflows/build-and-release.yml:99]: Hmm... If `pypi_password` is wrong the it means that we should fix this in other repos as from what I noticed we use `lowercase` variable in them

File: .github/actions/install/action.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  steps:
+    - name: Setup python additional packages
+      shell: bash
+      run: >
+        python -m pip install --upgrade pip
+        pip install setuptools wheel py coveralls
+
+    - name: Install requirements
+      shell: bash
+      run: >
```
Comment (benjamin-bergia) [inline .github/actions/install/action.yml]:
```suggestion
      run: pip install -e .[tests]
```

File: .github/actions/run-tests/action.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  using: composite
+  steps:
+    - name: Run isort
+      shell: bash
+      run: >
+        isort -rc -c -df
+
+    - name: Run pytest
+      shell: bash
+      run: >
```
Comment (benjamin-bergia) [inline .github/actions/run-tests/action.yml:13]:
```suggestion
      run: >
        py.test
        --pep8
        --cov=jsonschema2rst
        --cov-report=term-missing
        tests
        jsonschema2rst
```

File: .github/actions/install/action.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+name: Setup python environenment
+description: Setup python packages and requirements
+
+runs:
+  using: composite
+  steps:
+    - name: Setup python additional packages
+      shell: bash
+      run: >
```
Comment (benjamin-bergia) [inline .github/actions/install/action.yml]:
```suggestion
      run: |
```

File: .github/actions/install/action.yml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+name: Setup python environenment
```
Comment (benjamin-bergia) [inline .github/actions/install/action.yml:1]:
```suggestion
name: Setup python environment
```
Could you also name the directory so it matches?

**Review by benjamin-bergia — CHANGES_REQUESTED** [review]

File: .github/actions/install/action.yml (hunk 7 above)
Comment (pazembrz) [reply .github/actions/install/action.yml]: Already fixed

File: .github/actions/install/action.yml (hunk 9 above)
Comment (pazembrz) [reply .github/actions/install/action.yml]: Already fixed

File: .github/actions/install/action.yml (hunk 10 above)
Comment (pazembrz) [reply .github/actions/install/action.yml:1]: :+1:

**Review by benjamin-bergia — APPROVED** [review]

**Pull request merged**